Paopasatiht/iarepus-chat-bot
# Feature/pattern matching

## Step to run test
- 1.) Install the requirement.txt files
- 2.) Make sure that path to data_corpus.csv in `chat.py` and `utils/dialogue_manager.py` path are correct
- 3.) Run test by run python chat.py

### Refactor codebase
- move all of utility files onto utils folder
- move all file that related to Neural network to be under model folder

### Add feature matching pattern
- script `utils/dialogue_manager.py`, this script working by compare the similarity of the "query text" and "reference text" on the `data_corpus.csv` file. Find the topmost probability and pick the answer from csv file
- scipt `utils/pythainlp_utils.py` add feature "sentence embedding" and "sentence similarity"
- script `generate_q_vecctor.py` use to generate a Question_vector columns in data_corpus.csv
